### ISSUE TITLE
Difficulties to chain http requests

### ISSUE THREAD

**Issue by marvinroger** [node-red/node-red#582]
Hello,

I have a problem that appeared on 0.10.4, but was not present on 0.10.3.
Imagine you want to chain multiple HTTP requests, like so:

* HTTP endpoints
  * [GET] /1 which return 1
  * [GET] /2 which return 2
* Flow
  * `[GET] /1` ---> Debug msg.payload ---> `[GET] /2` ---> Debug msg.payload

The log should show 1, then 2. But on 0.10.4 it shows 1 then 1 again. It seems that the http request node changes the msg.url to the node url, meaning that the next http request node is overriden by the msg.url, which is why the same URL is called two times.

**Flow (assuming node-red listens on 127.0.0.1:1880)**

```
[{"id":"cce44c6b.35bd2","type":"inject","name":"","topic":"","payload":"","payloadType":"none","repeat":"","crontab":"","once":false,"x":94,"y":52,"z":"f48bce72.710dc","wires":[["dedd67ed.fbce28"]]},{"id":"4276fee6.f35ee8","type":"http in","name":"","url":"/1","method":"get","x":74,"y":153,"z":"f48bce72.710dc","wires":[["153d219e.58acce"]]},{"id":"3de3506e.153298","type":"http in","name":"","url":"/2","method":"get","x":81,"y":211,"z":"f48bce72.710dc","wires":[["56f2eec9.3a1488"]]},{"id":"153d219e.58acce","type":"change","action":"replace","property":"payload","from":"","to":"1","reg":false,"name":"","x":224,"y":153,"z":"f48bce72.710dc","wires":[["bdb133d3.fccd6"]]},{"id":"56f2eec9.3a1488","type":"change","action":"replace","property":"payload","from":"","to":"2","reg":false,"name":"","x":238,"y":212,"z":"f48bce72.710dc","wires":[["bdb133d3.fccd6"]]},{"id":"bdb133d3.fccd6","type":"http response","name":"","x":421,"y":174,"z":"f48bce72.710dc","wires":[]},{"id":"dedd67ed.fbce28","type":"http request","name":"request 1","method":"GET","ret":"txt","url":"http://127.0.0.1:1880/1","x":239,"y":55,"z":"f48bce72.710dc","wires":[["aaa42965.71cb1","ad93eba6.80b0a8"]]},{"id":"aaa42965.71cb1","type":"http request","name":"request 2","method":"GET","ret":"txt","url":"http://127.0.0.1:1880/2","x":527,"y":51,"z":"f48bce72.710dc","wires":[["ad93eba6.80b0a8"]]},{"id":"ad93eba6.80b0a8","type":"debug","name":"","active":true,"console":"false","complete":"payload","x":687,"y":116,"z":"f48bce72.710dc","wires":[]}]
```

**Flow with msg.url deletion between the two http request node, fixing the problem (assuming node-red listens on 127.0.0.1:1880)**

```
[{"id":"cce44c6b.35bd2","type":"inject","name":"","topic":"","payload":"","payloadType":"none","repeat":"","crontab":"","once":false,"x":94,"y":52,"z":"f48bce72.710dc","wires":[["dedd67ed.fbce28"]]},{"id":"4276fee6.f35ee8","type":"http in","name":"","url":"/1","method":"get","x":74,"y":153,"z":"f48bce72.710dc","wires":[["153d219e.58acce"]]},{"id":"3de3506e.153298","type":"http in","name":"","url":"/2","method":"get","x":81,"y":211,"z":"f48bce72.710dc","wires":[["56f2eec9.3a1488"]]},{"id":"153d219e.58acce","type":"change","action":"replace","property":"payload","from":"","to":"1","reg":false,"name":"","x":224,"y":153,"z":"f48bce72.710dc","wires":[["bdb133d3.fccd6"]]},{"id":"56f2eec9.3a1488","type":"change","action":"replace","property":"payload","from":"","to":"2","reg":false,"name":"","x":238,"y":212,"z":"f48bce72.710dc","wires":[["bdb133d3.fccd6"]]},{"id":"bdb133d3.fccd6","type":"http response","name":"","x":421,"y":174,"z":"f48bce72.710dc","wires":[]},{"id":"dedd67ed.fbce28","type":"http request","name":"request 1","method":"GET","ret":"txt","url":"http://127.0.0.1:1880/1","x":239,"y":55,"z":"f48bce72.710dc","wires":[["ad93eba6.80b0a8","553d50f.84f443"]]},{"id":"aaa42965.71cb1","type":"http request","name":"request 2","method":"GET","ret":"txt","url":"http://127.0.0.1:1880/2","x":588,"y":51,"z":"f48bce72.710dc","wires":[["ad93eba6.80b0a8"]]},{"id":"ad93eba6.80b0a8","type":"debug","name":"","active":true,"console":"false","complete":"payload","x":668,"y":132,"z":"f48bce72.710dc","wires":[]},{"id":"553d50f.84f443","type":"change","action":"delete","property":"url","from":"","to":"","reg":false,"name":"","x":438,"y":51,"z":"f48bce72.710dc","wires":[["aaa42965.71cb1"]]}]
```

I hope I was clear enough!

**Comment by dceejay** [comment]
Hi - apologies - was getting ahead of myself. Reverted the change.
idea was to pass on the inbound url (whatever that ended up being if you had used templating on the way in) - but the one in the node should have priority... however previous versions (0.9x) gave priority to the msg - and so we added a deprecated message - but of course that only just made it to the released version... so we can't really change the underlying behaviour - to give priority to the node url - until we have given people a decent warning period... probably release 11.

So the net is my change was premature... so reverting it until then...
